cli/npm/managed/resolvers/local/bin_entries.rs

```diff
@@ -3,7 +3,201 @@
 use crate::npm::managed::NpmResolutionPackage;
 use deno_core::anyhow::Context;
 use deno_core::error::AnyError;
+use deno_npm::resolution::NpmResolutionSnapshot;
+use deno_npm::NpmPackageId;
+use std::collections::HashMap;
+use std::collections::HashSet;
+use std::collections::VecDeque;
 use std::path::Path;
+use std::path::PathBuf;
+
+#[derive(Default)]
+pub(super) struct BinEntries {
+  /// Whether there's a collision in bin names
+  /// (i.e. two packages with the same bin name)
+  has_collision: bool,
```

Comment:
Perhaps instead of a bool we should store a list of package names with collissions? That way we can search for only those packages in the graph and not all of them?